File: _CodeTopics/LeetCode_contest/weekly/weekly2020/205/WA--205_1.py
class Solution(object):
    def modifyString(self, s):
        """
        :type s: str
        :rtype: str
        """
        
        s = list(s)
        length = len(s)
        if s[0] == '?':
            s[0] = 'a'
        for i in range(1, length):
            if s[i] == '?':
                if s[i-1] != 'a':
                    s[i] = 'a'
                else:
                    s[i] = 'b'
            else:
                if s[i] == s[i-1]:
                    if s[i-1] != 'a':
                        s[i] = 'a'
                    else:
                        s[i] = 'b'
        return ''.join(s)
    
"""
https://leetcode-cn.com/submissions/detail/105220841/

89 / 100 个通过测试用例
状态：解答错误

输入："j?qg??b"
输出："jaqgaba"
预期："jaqgacb"
"""

File: _CodeTopics/LeetCode/1401-1600/001576--weekly205_1/tran/WA--205_1.cpp
class Solution {
public:
    string modifyString(string s) {
        
        // 第 205 场周赛第一题
        // https://leetcode-cn.com/submissions/detail/105224224/

        int length = s.size();
        if (s[0] == '?')
            s[0] = 'a';
        for (int i = 1; i < length; ++i) {
            if (s[i] == '?') {
                if (s[i-1] != 'a')
                    s[i] = 'a';
                else
                    s[i] = 'b';
            } else {
                if (s[i] == s[i-1]) 
                    s[i-1] = 'c';
            }
        }
        return s;
    }
};

/*
https://leetcode-cn.com/submissions/detail/255048395/

775 / 776 个通过测试用例
状态：解答错误

输入：
"c?a"
输出：
"cca"
预期结果：
"cba"
*/
/*
这个就是之前对了，但是我今天看就有问题，果然确实是有问题的。
*/
